### PR TITLE
Add get_next_release_version and commit sha to update internal release semver

### PR DESCRIPTION
**Pull request by colin-rogers-dbt**
resolves #

Changes:
* Remove version_number since we infer it from the package 
* Adds option to skip-tests (debatable whether we should have this it was more to make it convenient to iterate on changes to the workflow)
* semver will now be `<major>.<minor>.<patch>.post<num>+<commit sha>`

### Checklist

- [ ] I have read [the contributing guide](https://github.com/dbt-labs/dbt-release/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [ ] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [ ] I have run this code in development and it appears to resolve the stated issue

### CODE REVIEW

**Comment by colin-rogers-dbt** [comment]
Successful run: https://github.com/dbt-labs/dbt-postgres/actions/runs/9556148717/job/26340845931

File: .github/actions/next-internal-version/action.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: "Get Next Internal Release Version"
```
Comment (mikealfare) [inline .github/actions/next-internal-version/action.yaml:1]: We're referring to these as Cloud releases going forward, right?

File: .github/actions/next-internal-version/action.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+description: "Determines if the published image should include `latest` tags"
+
+inputs:
+  version_number:
+    description: "SemVer of the package being published (i.e. 1.7.2, 1.8.0a1, etc.)"
+    required: true
+  versions_published:
+    description: "SemVer of the package being published (i.e. 1.7.2, 1.8.0a1, etc.)"
```
Comment (mikealfare) [inline .github/actions/next-internal-version/action.yaml]: Looks like a copy paste error.

File: .github/actions/next-internal-version/action.yaml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+name: "Get Next Internal Release Version"
+description: "Determines if the published image should include `latest` tags"
```
Comment (mikealfare) [inline .github/actions/next-internal-version/action.yaml]: Does this determine if `latest` should be used, what the next version is, both? This description doesn't appear to match the `outputs` on the surface.

File: .github/actions/next-internal-version/get_next_release_version.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+def main():
+    parser = argparse.ArgumentParser(description="Get the next release version")
+    parser.add_argument("--released_versions", type=str, help="comma delimited list of released versions")
+    parser.add_argument("--target_version", help="Target version to compare against")
+    args = parser.parse_args()
+    released_versions = list(filter(None, args.released_versions.split(",")))
+    target_version = args.target_version
+    latest_version = increment_latest_version(released_versions, target_version)
+    print(f"{latest_version.major}.{latest_version.minor}.{latest_version.patch}{latest_version.prerelease[0]}")
```
Comment (mikealfare) [inline .github/actions/next-internal-version/get_next_release_version.py:47]: `main` doesn't return anything. Is printing how we get the result back into the GHA environment?

File: .github/workflows/internal-archive-release.yml [hunk 5]
```diff
@@ -299,41 +290,48 @@ jobs:
...
           echo "[Debug] version_in_file: ${version_in_file}"
-          echo "[Debug] current_latest_version: ${current_latest_version}"
+          
+          echo "CURRENT_PKG_VERSION=${version_in_file}" >> "$GITHUB_ENV"
+          echo "VERSIONS_PUBLISHED=$(echo "${versions[*]}"| tr '\n' ',')" >> "$GITHUB_ENV"
+          echo "VERSION_FILE=${version_file}" >> "$GITHUB_ENV"
+
+      - name: "Determine next internal release version"
+        id: get_next_internal_release_version
+        uses: "dbt-labs/dbt-release/.github/actions/next-internal-version@addPythonScriptComparison"
```
Comment (mikealfare) [inline .github/workflows/internal-archive-release.yml:316]: Reminder to point this back at `main` before merging

File: .github/workflows/internal-archive-release.yml [hunk 6]
```diff
@@ -299,41 +290,48 @@ jobs:
...
+        with:
+          version_number: $CURRENT_PKG_VERSION
+          versions_published: $VERSIONS_PUBLISHED
 
+      - name: "Update version in package and setup.py"
+        run: |
+          internal_release_version="${{ steps.get_next_internal_release_version.outputs.internal_release_version }}"
+          commit_sha="$(git rev-parse HEAD)"
+          setup_file="./setup.py"
```
Comment (mikealfare) [inline .github/workflows/internal-archive-release.yml:325]: This is not true for `dbt-postgres` and `dbt-adapters`.

File: .github/actions/next-internal-version/get_next_release_version.py (hunk 4 above)
Comment (colin-rogers-dbt) [reply .github/actions/next-internal-version/get_next_release_version.py:47]: indeed it is

File: .github/workflows/internal-archive-release.yml (hunk 6 above)
Comment (colin-rogers-dbt) [reply .github/workflows/internal-archive-release.yml:325]: Good catch, I've added a link to this comment for our ticket to add dbt-adapters to the internal release pipeline: https://github.com/dbt-labs/adapters-team-private/issues/149

File: .github/actions/next-internal-version/action.yaml (hunk 1 above)
Comment (mikealfare) [reply .github/actions/next-internal-version/action.yaml]: I would also rename the action itself to match.

Comment (colin-rogers-dbt) [reply .github/actions/next-internal-version/action.yaml]: fair, also update the step name

**Review by mikealfare — APPROVED** [review]

**Pull request merged**